app/src/main/java/com/github/wnder/guessLocation/GuessFragment.java

```diff
@@ -109,9 +113,9 @@ private void onStyleLoaded(Style style){
         drawCircle(getActivity(), mapboxMap, cameraPosition);
     }
 
-    private void openPreviewActivity() {
+    private void openActivity(AppCompatActivity activity) {
```

Comment:
The signature should be changed to something like `public <T extends AppCompatActivity> void openActivity(Class<T> activity)` so that a class can be passed to it instead of an instance.

Reply:
done